File: examples/typescript/evm/getOrCreateAccount.ts
// Usage: pnpm tsx evm/getOrCreateAccount.ts

import { CdpClient } from "@coinbase/cdp-sdk";
import "dotenv/config";

const cdp = new CdpClient();

// Get or create an account
const name = "Account1";
const account = await cdp.evm.getOrCreateAccount({ name });
console.log("EVM Account Address: ", account.address);

const account2 = await cdp.evm.getOrCreateAccount({ name });
console.log("EVM Account 2 Address: ", account2.address);

const areAccountsEqual = account.address === account2.address;
console.log("Are accounts equal? ", areAccountsEqual);

const accountPromise1 = cdp.evm.getOrCreateAccount({ name: "Account" });
const accountPromise2 = cdp.evm.getOrCreateAccount({ name: "Account" });
const accountPromise3 = cdp.evm.getOrCreateAccount({ name: "Account" });
Promise.all([accountPromise1, accountPromise2, accountPromise3]).then(
  ([account1, account2, account3]) => {
    console.log("EVM Account Address 1: ", account1.address);
    console.log("EVM Account Address 2: ", account2.address);
    console.log("EVM Account Address 3: ", account3.address);
  }
);

File: examples/python/evm/get_or_create_account.py
# Usage: uv run python evm/get_or_create_account.py

import asyncio

from cdp import CdpClient
from dotenv import load_dotenv

load_dotenv()


async def main():
    async with CdpClient() as cdp:
        # Single account creation example
        name = "Account1"
        account = await cdp.evm.get_or_create_account(name=name)
        print("Account Address: ", account.address)
        account2 = await cdp.evm.get_or_create_account(name=name)
        print("Account 2 Address: ", account2.address)
        print("Are accounts equal? ", account.address == account2.address)

        # Concurrent account creation example
        account_coros = [
            cdp.evm.get_or_create_account(name="Account"),
            cdp.evm.get_or_create_account(name="Account"),
            cdp.evm.get_or_create_account(name="Account"),
        ]
        accounts = await asyncio.gather(*account_coros)
        for i, acc in enumerate(accounts, 1):
            print(f"EVM Account Address {i + 1}: ", acc.address)


asyncio.run(main())
